src/Components/Notes.js

```diff
@@ -1,22 +1,62 @@
-import React from "react";
+import React, { useState } from "react";
 import { useSelector } from "react-redux";
 import styled from "styled-components";
+import Button from "./Button";
 
 function Notes() {
   const notes = useSelector((state) => {
     return state.noteReducer.value;
   });
+
+  const [sorted, setSorted] = useState(false);
+
+  const sortNotes = (e) => {
+    e.preventDefault();
+    setSorted(!sorted);
+  };
+
+  const noteMap = () => {
+    let sortedData = [...notes];
+    if (sorted) {
+      sortedData = sortedData.sort((a, b) => (a.date > b.date ? 1 : -1));
+    } else {
+    }
+    return sortedData.map((n) => {
+      return (
+        <StickyNote key={n.id}>
+          <NoteDate>{n.date}</NoteDate>
+          <NoteName>{n.name}</NoteName>
+          <NoteContent>{n.note}</NoteContent>
+        </StickyNote>
+      );
+    });
+  };
+
   return (
     <div>
-      {notes.map((n) => {
-        return (
-          <StickyNote>
-            <NoteDate>{n.date}</NoteDate>
-            <NoteName>{n.name}</NoteName>
-            <NoteContent>{n.note}</NoteContent>
-          </StickyNote>
-        );
-      })}
+      {notes.length != 0 ? (
+        <Button
+          type="Sort"
+          value="Sort"
+          id="Sort"
+          onClick={sortNotes}
+          cssOptions={{
+            border: "none",
+            opacity: "0.8",
+            textdecor: "underline",
+            fontsize: "24px",
+            width: "10%",
+            float: "left",
+            background: "none",
+          }}
+        >
+          {sorted ? "Undo" : "Sort Notes"}
+        </Button>
+      ) : null}
+      {notes.length === 0 ? (
```

Comment:
You can use `{notes.length === 0 && <NoNotes> Notes will be displayed here </NoNotes>}` here

Reply:
Instead of two conditions here, why dont you have a ternary:

```
{notes.length === 0 ? (
<NoNotes> Notes will be displayed here </NoNotes>
) : (
<Button
          type="Sort"
          value="Sort"
          id="Sort"
          onClick={sortNotes}
          cssOptions={{
            border: "none",
            opacity: "0.8",
            textdecor: "underline",
            fontsize: "24px",
            width: "10%",
            float: "left",
            background: "none",
          }}
        >
          {sorted ? "Undo" : "Sort Notes"}
        </Button>
)}
```